transformations.py

```diff
@@ -0,0 +1,14 @@
+def mean(table, column):
+    return table.groupby('device_id')[column].mean()
+
+
+def most_common(table, column):
+    return table.groupby('device_id')[column].mode().iloc[0]
+
+
+transformations = {
+    'battery': {
+        'battery_level': mean,
```

Comment:
One problem with this approach is that it might be a little tricky to handle functions that take multiple columns as parameters. One slightly overwrought solution might be to use nested functions:

```
def process_multiple_columns(*columns):
  def _inner(table):
    # do something with multiple columns on this table
  return _inner

transformations = {
  'battery': [
    process_multiple_columns('column1', 'column2')
  ]
}

for transform in transformations['battery']:
  transform(battery_table)
```

Reply:
Another option would be to use a tuple to store the columns to transform on in the table, something like:

```
'battery': {
    ('battery_level', 'timestamp') : multi_column_transform,
    'battery_level' : mean, 
    ...
}
```

That would put the onus on this table of transforms to be correctly specified though, not sure if that is a good or bad thing. Something along the lines of your `process_multiple_columns` wrapper function could work as well.